server/task.go

```diff
@@ -457,6 +457,64 @@ func (s *Server) composeTaskRelationship(ctx context.Context, task *api.Task) er
 	return nil
 }
 
+func (s *Server) composeTaskRelationshipValidateOnly(ctx context.Context, task *api.Task) error {
```

Comment:
Could u please elaborate on the overall plan, because this looks like the same as composeTaskRelationship.

Reply:
If the end result is we will have 2 separate versions for validateOnly and non-validateOnly, then our code is probably broken.

@d-bytebase I remembered we had a discussion originally when introducing the validateOnly path, and it's a temporary hack to have validateOnly and non-validateOnly code during the transition of moving issue/pipeline creation to the backend.

Ideally, the code should be like 

createPipeline() {
  validate()
  if !validateOnly {
   doActualWork()
  }
}

So most code should be reused.

Reply:
The pipeline generation move to the backend is already completed and cleaned up. The validate_only flag is meant to return the expected response without actually creating side-effects such as database data updates. 

    https://google.aip.dev/163

doWithSideEffect() composition should do XRaw to X conversion. doWithoutSideEffects() doesn't need it, we would have to branch code on the composition part, hopefully for now. I'm proposing to create a composition layer later, and hopefully, these two code paths can converge again at the time. This change is meant to simply unblock the entity refactoring.

```
createPipeline() {
  if validateOnly {
    return doWithoutSideEffects()
  }
  return doWithSideEffect()
}
```

Reply:
SGTM since we have a plan.